### PR TITLE
feat(bridge): add bridge flow

### PR DESCRIPTION
**Pull request by LuizAsFight**
Closes [FRO-10](https://linear.app/fuel-network/issue/FRO-10/assembly-eth-deposit-feature)
Closes [FRO-12](https://linear.app/fuel-network/issue/FRO-12/deposit-processing-steps-component)

include bridge flow, machines and hooks needed.

- use integrated FuelWallet
  - refact/fix some things in integration
- change eth node development port to 8545
- chains
  - automatically use correct chains in development/test environments
  - utils for handling fuel/eth chains
- use integrated ETH wallet
- created independent BridgeTabs container
- created independent BridgeButton container
  - (UX) "Bridge Button" shows flow text and also allows to trigger necessary actions to continue in flow
    - for example "Connect ETH wallet", "Connect Fuel Wallet", "Pick your asset" etc...
- bump fuel-ui to use new AmountInput and new branding
- fix some remaining theming issues
- in DEVELOPMENT mode, get FuelMessageContractId from localhost:8080 and set in ENV variable
- create new system for Chains
  - eth / fuel
- Transaction Popup with STEPS will show up after starting bridge
  - popup is not complete yet, will be done in next PRs when doing the transaction popup itself

### CODE REVIEW

**Review by luizstacio — CHANGES_REQUESTED** [review]
- When waiting for connect ETH it's showing `Connect Foundry Wallet` the button should have a fixed name not based on the name of the network but on the type of the network.
- Move the config of the network to a env variable. On the build it should be pointing to the Goerli testnet to enable test on the live preview.
- After fill the amount on input and clean the input the Button don't return to the previous status and is enable to click. It should return to the previous state `Type amount to transfer`.
- The transaction history is not centralizing the TX Component it should have the same padding on all the sides
- After submit the TX with success the amount input should be cleared.

File: packages/app/src/systems/Chains/eth/utils/logo.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import ethLogoSrc from '../../../../../public/eth-logo.svg';
```
Comment (luizstacio) [inline packages/app/src/systems/Chains/eth/utils/logo.ts:1]: You can use `~public/` and `~/<folder>` to avoid `../.../.......`

**Review by luizstacio — CHANGES_REQUESTED** [review]
- When waiting for connect ETH it's showing `Connect Foundry Wallet` the button should have a fixed name not based on the name of the network but on the type of the network.
- Move the config of the network to a env variable. On the build it should be pointing to the Goerli testnet to enable test on the live preview.
- After fill the amount on input and clean the input the Button don't return to the previous status and is enable to click. It should return to the previous state `Type amount to transfer`.
- The transaction history is not centralizing the TX Component it should have the same padding on all the sides
- After submit the TX with success the amount input should be cleared.

File: package.json [hunk 2]
```diff
@@ -38,9 +39,9 @@
     "xstate:typegen": "pnpm -r xstate:typegen"
   },
   "dependencies": {
-    "@fuel-ui/config": "0.15.0-dev-c6f1a1b",
-    "@fuel-ui/css": "0.15.0-dev-c6f1a1b",
-    "@fuel-ui/react": "0.15.0-dev-c6f1a1b"
+    "@fuel-ui/config": "0.15.0-dev-9c7ba65",
```
Comment (matt-user) [inline package.json:42]: can the fuel-ui deps be pinned to a non-dev version yet?  I guess that should be done in a separate pr

File: packages/app/load.envs.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+function getEnvName() {
+  if (process.env.NODE_ENV === 'production') {
+    return '.env.production';
+  }
+  if (process.env.NODE_ENV === 'test') {
+    return '.env.test';
+  }
+}
+
+function getEthFuelL1Contracts() {
```
Comment (matt-user) [inline packages/app/load.envs.js:24]: We should add this to the env and not fetch IMO

File: packages/app/load.envs.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+// Export the version to be used on database
+// and application level
+const versions = getVersion();
+process.env.VITE_APP_VERSION = versions.version;
+
+// Export ETH Fuel contracts addresses
+const ethFuelContracts = getEthFuelL1Contracts();
+if (ethFuelContracts && ethFuelContracts.FuelMessagePortal) {
+  process.env.VITE_ETH_FUEL_MESSAGE_PORTAL = ethFuelContracts.FuelMessagePortal;
```
Comment (matt-user) [inline packages/app/load.envs.js:54]: oh wait I guess we do add this to the env nvm

File: packages/app/src/systems/Chains/fuel/containers/FuelInstallDialog.tsx [hunk 5]
```diff
@@ -1,10 +1,10 @@
 import { cssObj } from '@fuel-ui/css';
 import { Button, Dialog, IconButton, Icon } from '@fuel-ui/react';
 
-import { useAccountConnectionFuel } from '~/systems/Accounts';
+import { useFuelAccountConnection } from '../hooks';
 
-export const FuelInstallPopup = () => {
-  const { handlers } = useAccountConnectionFuel();
+export const FuelInstallDialog = () => {
```
Comment (matt-user) [inline packages/app/src/systems/Chains/fuel/containers/FuelInstallDialog.tsx:6]: This no longer displays when the user does not have a fuel wallet installed

File: packages/app/src/systems/Chains/eth/services/txEthToFuel.ts [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+    const receipt = await ethTx.wait();
+    const event = fuelPortal.interface.parseLog(receipt.logs[0]);
+    const depositNonce = bn(event.args.nonce.toHexString());
+
+    return depositNonce;
+  }
+
+  static async getFuelMessage(input: TxEthToFuelInputs['getFuelMessage']) {
+    if (!input?.ethTxNonce) {
+      throw new Error('No nonce informed');
```
Comment (matt-user) [inline packages/app/src/systems/Chains/eth/services/txEthToFuel.ts:93]:
```suggestion
      throw new Error('No nonce found');
```

Informed sounds a little weird here to me.  I think found is a better fit.

File: packages/app/src/systems/Chains/eth/services/txEthToFuel.ts [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+    }
+    if (!input?.fuelProvider) {
+      throw new Error('No provider for Fuel informed');
```
Comment (matt-user) [inline packages/app/src/systems/Chains/eth/services/txEthToFuel.ts:96]:
```suggestion
      throw new Error('No provider for Fuel found');
```

File: packages/app/src/systems/Chains/eth/services/txEthToFuel.ts [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+    }
+    if (!input?.fuelAddress) {
+      throw new Error('No address for Fuel informed');
```
Comment (matt-user) [inline packages/app/src/systems/Chains/eth/services/txEthToFuel.ts:99]:
```suggestion
      throw new Error('No address for Fuel found');
```

File: packages/app/src/systems/Chains/fuel/chains.ts [hunk 9]
```diff
@@ -0,0 +1,13 @@
+export const fuelDev = {
```
Comment (matt-user) [inline packages/app/src/systems/Chains/fuel/chains.ts:1]: Should we consider typing this data?

File: packages/app/src/systems/Chains/fuel/hooks/useFuelAccountConnection.tsx [hunk 10]
```diff
@@ -4,16 +4,25 @@ import {
...
+export const useFuelAccountConnection = () => {
   const fuel = useFuel();
   const { account } = useAccount();
   const { isConnected } = useIsConnected();
-  const { connect, error, isLoading } = useConnect();
+  const { connect, error, isLoading: isConnecting } = useConnect();
   const { disconnect } = useDisconnect();
+  const { provider } = useProvider();
+
+  const address = useMemo(
```
Comment (matt-user) [inline packages/app/src/systems/Chains/fuel/hooks/useFuelAccountConnection.tsx:22]: Do we gain efficiency from this being in a useMemo?  It seems to me this could just be a variable initialization with no hooks.

File: packages/app/src/systems/Chains/fuel/utils/chain.ts [hunk 11]
```diff
@@ -0,0 +1,5 @@
+import type { SupportedChain } from '../../types';
+
+export const isFuelChain = (chain: SupportedChain | undefined | null) => {
```
Comment (matt-user) [inline packages/app/src/systems/Chains/fuel/utils/chain.ts:3]:
```suggestion
export const isFuelChain = (chain?: SupportedChain | null) => {
```

would this be equivalent?

File: packages/app/src/systems/Chains/fuel/utils/logo.ts [hunk 12]
```diff
@@ -0,0 +1,3 @@
+import fuelLogoSrc from '../../../../../public/fuel-logo.svg';
+
+export { fuelLogoSrc };
```
Comment (matt-user) [inline packages/app/src/systems/Chains/fuel/utils/logo.ts:3]: Should we instead use the `FuelLogo` from the fuel-ui library everywhere?

File: scripts/brigde/src/utils/wallets.ts [hunk 13]
```diff
@@ -3,7 +3,7 @@ import type { WalletUnlocked } from 'fuels';
 import { Provider, Wallet } from 'fuels';
 
 export const ethProvider = new ethers.providers.JsonRpcProvider(
-  'http://localhost:9545'
+  'http://localhost:8545'
```
Comment (matt-user) [inline scripts/brigde/src/utils/wallets.ts:6]: I've seen the 8545 port a couple of times.  Is it worth it to put it in a variable?

**Review by matt-user — CHANGES_REQUESTED** [review]

File: packages/app/load.envs.js (hunk 4 above)
Comment (LuizAsFight) [reply packages/app/load.envs.js:54]: yeah so in dev we get com L1 ETH docker we run. for test we will use a fixed one from goerli network(.env.example)

File: packages/app/src/systems/Chains/fuel/hooks/useFuelAccountConnection.tsx (hunk 10 above)
Comment (LuizAsFight) [reply packages/app/src/systems/Chains/fuel/hooks/useFuelAccountConnection.tsx:22]: yes we do, will only process Address.fromString when `account` gets changed

File: scripts/brigde/src/utils/wallets.ts (hunk 13 above)
Comment (LuizAsFight) [reply scripts/brigde/src/utils/wallets.ts:6]: it's only inside `scripts` package, which will be removed when withdraw gets implemented

Comment (LuizAsFight) [reply scripts/brigde/src/utils/wallets.ts:6]: for network connection we use `foundry` variable from wagmi that already contains URL with correct 8545 port

**Comment by LuizAsFight** [comment]
> * The transaction history is not centralizing the TX Component it should have the same padding on all the sides


this is part of [FRO-17](https://linear.app/fuel-network/issue/FRO-17/create-transaction-view-popup-component). I just started the popup in this PR so we can see the transaction working. but all other details should be done in the PR correspondent to that popup component.

**Comment by LuizAsFight** [comment]
@matt-user @luizstacio thanks for the great reviews. I addressed everything, could you check again ?

File: packages/app/src/systems/Bridge/components/BridgeSteps/BridgeSteps.tsx [hunk 14]
```diff
@@ -87,7 +87,7 @@ const styles = {
     color: '$intentsBase12',
   }),
   icon: cssObj({
-    color: '$intentsBase12',
+    color: '$black12',
```
[the remark this answers is not in the document]
Comment (LuizAsFight) [reply packages/app/src/systems/Bridge/components/BridgeSteps/BridgeSteps.tsx:90]: I fixed as $black12 on purpose because it was not looking good in black theme. you can finish fixing it in the PR to create Tx Popup

**Review by matt-user — APPROVED** [review]

**Review by luizstacio — APPROVED** [review]
I was able to bridge using Goerli! ⚡️

I have created 3 issues related to this PR with some bugs and to add E2E. I'm approving to unblock other tasks but we should focus on it after.
